### ISSUE TITLE
Incorrect Calculation of the 4th Angle in sp3 Hybridization

### ISSUE THREAD

**Issue by TharunKumarrA** [TharunKumarrA/Molecule-Visualiser-DSA#4]
# Issue: Fix Miscalculation of the Fourth Angle in sp3 Hybridization

## Description
The current implementation for **sp3 (tetrahedral)** hybridization miscalculates the fourth bond angle. The resulting geometry does not align with the standard **tetrahedral geometry**, where all bond angles should be approximately **109.5°**.

## Impact
- The visual representation of **sp3 hybridized molecules** is incorrect.
- It affects the accuracy of molecular geometries such as **CH₄ (methane)**.
- This inconsistency impacts downstream calculations and visualizations.

## Acceptance Criteria
- The fourth bond angle is corrected to align with **tetrahedral geometry**.
- All bond angles in the sp3 configuration are approximately **109.5°**.
- The corrected geometry is validated against standard molecular geometries.

## Suggested Approach
1. Investigate the logic in the `getCoordinates()` function in **`Molecule.js`**.
2. Update the calculation method for the fourth bond direction to ensure **tetrahedral geometry** is achieved.
3. Use trigonometric and vector models to calculate the correct bond angles and directions for sp3 hybridization.

---

### File: `/src/components/Molecule.js`
- **Update Functionality:**
  - Modify the `getCoordinates()` function to fix the calculation for the fourth bond angle in sp3 hybridization.

---

### In `Molecule.js`
Identify the logic for **sp3 hybridization** in `getCoordinates()`:
```javascript
if (currentAtom.hybridisation === "sp3") {
  // Fix the fourth bond angle calculation
  // Normalize the vectors to maintain consistent bond length
  // Calculate the coordinates for each bond
  // Store or utilize bondCoordinates
}
```

**Comment by SaranHiruthikM** [comment]
Heyy!! I would like to take up the issue. Please assign it to me @TharunKumarrA

**Comment by TharunKumarrA** [comment]
Yes. Assigned @hirux06

**Comment by TharunKumarrA** [comment]
Yeah sorry for the delayed reply. I'm kinda not well, so I'll discuss it tomorrow if it works

**Comment by SaranHiruthikM** [comment]
Hey @TharunKumarrA !! I’m sorry, but I’m unable to work on the assigned issue due to time constraints and challenges in getting the expected output. I kindly request you to unassign me so that it can be reassigned to someone else. Sorry for taking so much time.
Thanks

**Comment by TharunKumarrA** [comment]
Okay sure saran. Thanks for trying it out
